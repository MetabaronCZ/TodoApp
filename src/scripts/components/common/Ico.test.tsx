import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { Ico } from 'components/common/Ico';

import { icos } from 'modules/icos';
import { theme } from 'modules/theme';
import { hexToRgb } from 'modules/colors';
import { withMockedProviders } from 'test/component';

describe('components/common/Ico', () => {
  it('should render correctly', () => {
    const tree = render(withMockedProviders(<Ico ico="close" />));
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByText(icos.close)).toBeInTheDocument();
    expect(tree.container.querySelector('i')).toBeInTheDocument();
  });

  it('should render with props', () => {
    const { container } = render(
      withMockedProviders(<Ico ico="close" color="error" />),
    );
    expect(container).toMatchSnapshot();

    expect(container.firstChild).toHaveStyle({
      color: hexToRgb(theme.color.error),
    });
  });
});
