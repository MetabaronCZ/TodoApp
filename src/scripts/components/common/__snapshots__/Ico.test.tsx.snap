// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/common/Ico > should render correctly 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <i
    class="c0"
  >
    ✖
  </i>
</div>
`;

exports[`components/common/Ico > should render with props 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <i
    class="c0"
    style="color: rgb(204, 34, 34);"
  >
    ✖
  </i>
</div>
`;
